src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -136,17 +127,70 @@ void CPlayer::drawDeathEffect(HexagonGame& mHexagonGame)
     }
 }
 
-void CPlayer::playerSwap(HexagonGame& mHexagonGame, bool mPlaySound)
-{
+void CPlayer::playerSwap(HexagonGame &mHexagonGame, bool mPlaySound) {
     angle += ssvu::pi;
     mHexagonGame.runLuaFunctionIfExists<void>("onCursorSwap"); // TODO: docs
 
-    if(mPlaySound)
-    {
+    if (mPlaySound) {
         mHexagonGame.getAssets().playSound("swap.ogg");
     }
 }
 
+[[nodiscard]] sf::Vector2f CPlayer::getPosition() const {
+    return pos;
+}
+
+[[nodiscard]] sf::Vector2f CPlayer::getStartPosition() const {
+    return startPos;
+}
+
+void CPlayer::kill(HexagonGame &mHexagonGame, bool mEffect) {
```

Comment:
What's the point of `mEffect`? Is there anywhere when this should be called with `mEffect=false`? Also, the name is not very self-explanatory. What "effect" are we talking about?

Reply:
wanted to use that function to replace standard methods of killing s player to simplify it.
And by the effect it means the death effect itself... Prob had to name it like this.

Reply:
also yes there are places where it supposed to be called with `mEffect=false` like for example while `tutorial mode` is on iirc.

Reply:
also yes there are places where it supposed to be called with `mEffect=false` like for example while `tutorial mode` is on iirc.